packages/icons/scripts/icons.js

```diff
@@ -1,35 +1,58 @@
 const fs = require("fs");
-const path = require("path");
 
 const { pascalCase } = require("case-anything");
 const dotenv = require("dotenv");
 const { fetchImages } = require("figma-tools");
 const prettier = require("prettier");
-const svgtojsx = require("svg-to-jsx");
+const svgToJsx = require("svg-to-jsx");
+const { optimize } = require("svgo");
 
 dotenv.config({ path: "../../.env" });
 
-const pretterOptions = JSON.parse(fs.readFileSync("../../.prettierrc", "utf-8"));
-
 console.log("Fetching icons...");
 
 fetchImages({
   fileId: "ASas6u2DMihEEzw8jPT1XC",
   format: "svg",
+  onEvent: event => console.log(`${event.type}: ${event.status}`),
   filter: component => component.pageName === "0.3 Icons",
 }).then(async svgs => {
   console.log("Building icons...");
 
-  const jsx = await Promise.all(svgs.map(svg => svgtojsx(svg.buffer)));
-  const data = svgs
-    .map((svg, index) => `export const ${pascalCase(svg.name)}Icon = () => ${jsx[index]}\n`)
+  const optimizedSvgs = await Promise.all(
+    svgs.map(svg =>
+      optimize(svg.buffer.toString(), {
+        plugins: [
+          {
+            name: "removeAttrs",
+            params: {
+              attrs: "(fill|stroke)",
+            },
+          },
+        ],
+      })
+    )
+  );
+  const jsxSvgs = await Promise.all(optimizedSvgs.map(svg => svgToJsx(svg.data)));
+  const allIconExports = ['import { SVGProps } from "react"']
+    .concat(
+      svgs.map((svg, index) => {
+        const name = `${pascalCase(svg.name)}Icon`;
+        const jsx = jsxSvgs[index]
+          /** Spread props for controlling the svg element. */
+          .replace('">\n', '" {...props}>\n');
```

Comment:
In the future, we can add babel or some simple AST parsing to do linting and more reliable transformations like this.